geode-core/src/test/java/org/apache/geode/internal/cache/SingleThreadJTAExecutorTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache;
+
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import org.awaitility.Awaitility;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InOrder;
+
+import org.apache.geode.CancelCriterion;
+
+public class SingleThreadJTAExecutorTest {
+  private SingleThreadJTAExecutor singleThreadJTAExecutor;
+  private TXState txState;
+  private ExecutorService executor;
+  private BeforeCompletion beforeCompletion;
+  private AfterCompletion afterCompletion;
+  private CancelCriterion cancelCriterion;
+
+  @Before
+  public void setup() {
+    txState = mock(TXState.class, RETURNS_DEEP_STUBS);
+    executor = Executors.newSingleThreadExecutor();
+    beforeCompletion = mock(BeforeCompletion.class);
+    afterCompletion = mock(AfterCompletion.class);
+    cancelCriterion = mock(CancelCriterion.class);
+    singleThreadJTAExecutor = new SingleThreadJTAExecutor(beforeCompletion, afterCompletion);
+  }
+
+  @Test
+  public void executeBeforeCompletionCallsDoOps() {
+    singleThreadJTAExecutor.executeBeforeCompletion(txState, executor, cancelCriterion);
+
+    InOrder inOrder = inOrder(beforeCompletion, afterCompletion);
```

Comment:
Move this up to the first line of the method. It should be before we call "executeBeforeCompletion".

Reply:
"this" refers to the "InOrder" line.